src/app/routes/registration/request-claim/request-claim.component.ts

```diff
@@ -178,83 +215,59 @@ export class RequestClaimComponent implements OnInit, SubjectElements {
   }
 
   async submit() {
+    if (!this.enrolmentForm.valid && !this.isRegistrationTypeSelected()) {
+      this.toastr.error('Enrolment Form is invalid.', TOASTR_HEADER);
+      return;
+    }
     this.loadingService.show();
-    if (this.enrolmentForm.valid) {
-      let did = undefined;
-      if (this.selectedRole.issuer) {
-        if (this.selectedRole.issuer.roleName) {
-          // Retrieve list of issuers by roleName
-          did = await this.iamService.iam.getRoleDIDs({
-            namespace: this.selectedRole.issuer.roleName
-          });
-        } else if (this.selectedRole.issuer.did) {
-          did = this.selectedRole.issuer.did;
-        }
+    let did;
+    if (this.selectedRole.issuer) {
+      if (this.selectedRole.issuer.roleName) {
+        // Retrieve list of issuers by roleName
+        did = await this.iamService.iam.getRoleDIDs({
+          namespace: this.selectedRole.issuer.roleName
+        });
+      } else if (this.selectedRole.issuer.did) {
+        did = this.selectedRole.issuer.did;
       }
+    }
 
-      if (did && did.length) {
-        this.submitting = true;
-        this.loadingService.show('Please confirm this transaction in your connected wallet.');
-
-        try {
-          // Construct Fields
-          let fields = [];
-          let values = this.enrolmentForm.value.fields;
-          for (let i = 0; i < this.fieldList.length; i++) {
-            fields.push({
-              key: this.fieldList[i].label,
-              value: values[i]
-            });
-          }
-
-          // Submit first digit of version
-          // because legacy role's had version format of '1.0.0'
-          // but we version should be persisted as numbers
-          const parseVersion = (version: string | number) => {
-            if (typeof (version) === 'string') {
-              return parseInt(version.split('.')[0], 10);
-            }
-            return version;
-          };
-          const claim = {
-            fields: JSON.parse(JSON.stringify(fields)),
-            claimType: this.selectedNamespace,
-            claimTypeVersion: parseVersion(this.selectedRole.version) || DEFAULT_CLAIM_TYPE_VERSION
-          };
-
-          await this.iamService.iam.createClaimRequest({
-            issuer: did,
-            claim,
-            subject: this.roleTypeForm.value.assetDid ? this.roleTypeForm.value.assetDid : undefined
-          });
-
-        } catch (e) {
-          console.error('Enrolment Failed', e);
-          this.toastr.error(e, TOASTR_HEADER);
-          this.submitting = false;
-        } finally {
-          this.loadingService.hide();
-        }
-
-        this.displayAlert('Request to enrol as ' + this.roleTypeForm.value.roleType.name.toUpperCase() + ' is submitted for review and approval.',
-          'success');
-      } else {
-        this.toastr.error('Cannot identify issuer for this role.', TOASTR_HEADER);
+    if (did && did.length) {
```

Comment:
So i can send `issuer: undefined` in `createClaimRequest`? Because this property is right now required. Or maybe we should change types in iam-client-lib ?

Reply:
Hmm, yes, maybe we should be able to see `issuer: undefined` in `createClaimRequest`. But maybe for now, for the "on-chain", I would say it is fine to just leave the array empty.